### PR TITLE
Implement Blacklight-Gallery

### LINKED ISSUE #139: Try Out Blacklight-Gallery

### PR DESCRIPTION
**Pull request by terrellt**
This implements all of blacklight gallery. Likely we'll want to handle this on a
per-set basis, and probably only parts of it.

Closes #139

### CODE REVIEW

**Comment by terrellt** [comment]
This implements a gallery, a masonry, and a slideshow view for all items. The question is: do we want all of these? Do we want these in the default view at all? Just for some sets?

**Comment by wickr** [comment]
Hard to say until more people see them in action. I think leaving all of those as default options is fine for now.

**Comment by terrellt** [comment]
@wickr That makes sense to me.

File: app/models/blacklight_config.rb [hunk 1]
```diff
@@ -24,7 +24,8 @@ def configurations
       DefaultConfiguration.new(configuration),
       SearchFieldConfiguration.new(configuration),
       ShowActions.new(configuration),
-      ThumbnailConfiguration.new(configuration)
+      ThumbnailConfiguration.new(configuration),
+      Gallery.new(configuration)
```
Comment (straleyb) [inline app/models/blacklight_config.rb]: Just a question. Do we want to keep the names of these consistent?  _____Configuration. Or does it really matter?

File: app/values/solr_document_derivative_paths.rb [hunk 2]
```diff
@@ -2,6 +2,10 @@ class SolrDocumentDerivativePaths
   pattr_initialize :document
   delegate :[], :to => :derivative_paths
 
+  def [](key)
+    derivative_paths[key] || NullDerivativePath.instance
```
Comment (straleyb) [inline app/values/solr_document_derivative_paths.rb]: Null Object Pattern ftw :+1:

File: app/models/blacklight_config.rb (hunk 1 above)
Comment (terrellt) [inline app/models/blacklight_config.rb]: I'm ambivalent, because of their namespacing I'm even tempted to kill Configuration from the other objects.

**Pull request merged**